### PR TITLE
Send metals/didFocusTextDocument notification on buffer change.

### PR DESCRIPTION
**Pull request by kurnevsky**
https://scalameta.org/metals/docs/editors/new-editor.html#metalsdidfocustextdocument

Should be sent for decorations to work properly: https://github.com/scalameta/metals/issues/1258#issuecomment-578398187

### CODE REVIEW

File: lsp-metals.el [hunk 1]
```diff
@@ -183,6 +183,15 @@ Should be ignored if there is no open doctor window."
                         (c (ignore (lsp-warn "Unknown metals client command: %s" c))))))
     (apply command (append (list workspace) (ht-get params "arguments") nil))))
 
+(defvar lsp-metals--current-buffer nil
+  "Current buffer used to send `metals/didFocusTextDocument' notification.")
+
+(defun lsp-metals--did-fucus ()
```
Comment (yyoncho) [inline lsp-metals.el]: typo: focus

File: lsp-metals.el [hunk 2]
```diff
@@ -183,6 +183,15 @@ Should be ignored if there is no open doctor window."
...
+  "Send `metals/didFocusTextDocument' when current buffer changes."
+  (when (not (eq lsp-metals--current-buffer (current-buffer)))
```
Comment (yyoncho) [inline lsp-metals.el]: nit: instead of `(when (not` use `(unless`

File: lsp-metals.el [hunk 3]
```diff
@@ -183,6 +183,15 @@ Should be ignored if there is no open doctor window."
                         (c (ignore (lsp-warn "Unknown metals client command: %s" c))))))
     (apply command (append (list workspace) (ht-get params "arguments") nil))))
 
+(defvar lsp-metals--current-buffer nil
```
Comment (kurnevsky) [inline lsp-metals.el]: Maybe better to store it in workspace metadata?

Comment (yyoncho) [reply lsp-metals.el]: Is it working fine with this version of the code?

Comment (kurnevsky) [reply lsp-metals.el]: Yes, but it will send unnecessary notifications when switching between different metals workspaces.

Comment (yyoncho) [reply lsp-metals.el]: Before sending a message you may check if the buffer is managed by metals instance. As a side note. E. g.
```
(with-current-buffer "HelloSpec.scala"
  (lsp--workspace-buffers (lsp-find-workspace 'metals (buffer-file-name)) ))
```

Comment (kurnevsky) [reply lsp-metals.el]: I add local hooks only to metals buffers, so no need to check it.

Comment (yyoncho) [reply lsp-metals.el]: So IIUC 

> it will send unnecessary notifications when switching between different metals workspaces.

is not a problem?

Comment (kurnevsky) [reply lsp-metals.el]: Yes, not a problem at all :)

**Pull request merged**